### PR TITLE
Device groups: Create DB structure to "groups".

### PR DESCRIPTION
**Pull request by ldjebran**
# Description

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

Fixes # THEEDGE-1590

## Type of change

What is it?

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Documentation update
- [ ] Tests update
- [ ] Refactor

# Checklist:

- [X] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [X] I have added tests that prove my fix is effective or that my feature works
- [X] New and existing unit tests pass locally with my changes
- [X] I run `go fmt ./...` to check that my code is properly formatted
- [X] I run `go vet ./...` to check that my code is free of common Go style mistakes

### CODE REVIEW

File: pkg/models/groups.go [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+
+import (
+	"errors"
+	"regexp"
+)
+
+// Group is a record of Edge Devices Groups
+// Account is the account associated with the group
+// Type is the group type and must be "static" or "dynamic"
+type Group struct {
```
Comment (biancarosa) [inline pkg/models/groups.go:11]: Maybe we should call it DeviceGroups?

Comment (ldjebran) [reply pkg/models/groups.go:11]: I was also thinking this way, but there is also devices_groups , so then we will have too much devices groups and will became confusing , 
so we have devices_groups table for the collection 
and if this one will be DeviceGroup  we will have a device_groups 
naming is hard :)

we can rename but we will be confused with device_groups and devices_groups.
what do you think ?

Comment (ldjebran) [reply pkg/models/groups.go:11]: Or rename this to DeviceGroup   and the collection to device_groups_devices , what do you think ?

Comment (ldjebran) [reply pkg/models/groups.go:11]: anyway going to rename to DeviceGroup and the collection to device_groups_devices

seems more logical

**Review by biancarosa — APPROVED** [review]
LGTM!

**Pull request merged**